freud/common.py

```diff
@@ -10,8 +10,8 @@
 logger = logging.getLogger(__name__)
 
 
-def convert_array(array, dimensions, dtype=None,
-                  contiguous=True, array_name=None):
+def convert_array(array, dimensions, dtype=np.float32,
+                  contiguous=True):
```

Comment:
I'm inclined to require `contiguous=True` in all cases, and remove this option from the function signature. We cannot use arrays in C++ unless they're contiguous, and all previous calls to `convert_array` set this to `True`.